File: cpp/potega.cpp
/*
 * potega.cpp
 * 
 * a0 = 1
 * a1 = a
 * an = a * ... * a (n-czynników) dla n zaw. N+ - {1}
 */


#include <iostream>

using namespace std;

float potega_it(float x, int n)
{
    float wynik = 1;
    for (int i = 1; i <= n; i++)
    {
        wynik = wynik * x;
    }
    return wynik;
}

int main(int argc, char **argv)
{
    int n;
    float x;
    cout << "Podaj podstawe potêgi: " << endl;
    cin >> x;
    cout << "Podaj wyk³adnik potêgi: " << endl;
    cin >> n;
    cout << "Potêga: " << potega_it(x, n) <<endl;
	return 0;
}


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: python/potega.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.py


def potega_it(podstawa, wykladnik):
    wynik = 1
    for i in range(0, wykladnik):
        wynik = wynik * podstawa
    return wynik

# potega_rek(a, 0) = 1 dla a różnego od 0
# potega_rek(a, n) = potega_rek(a, n - 1) * a dla n = N+


def potega_rek(a, n):
    if n == 0:
        return 1
    return potega_rek(a, n - 1) * a


def main(args):
    podstawa = float(input("Podaj podstawe potegi: "))
    wykladnik = int(input("Podaj wykladnik potegi: "))
    print("Wynik: ", potega_it(podsawa, wykładnik))
    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
